samples/CppArExample/qml/main.qml

```diff
@@ -45,31 +45,81 @@ ApplicationWindow {
+        visible: false
+        onTriggered: arSample.updateOriginCamera(latitude, longitude, altitude, heading);
+    }
+
     SettingsWindow {
         anchors {
             top: parent.top
             left: parent.left
             margins: 5
         }
 
+        // tracking options
         onStartTrackingClicked: arcGISArView.startTracking();
         onStopTrackingClicked: arcGISArView.stopTracking();
         onResetTrackingClicked: arcGISArView.resetTracking();
-        // onCalibrationClicked: not implemented
+        onCalibrationClicked: calibrationView.visible = ! calibrationView.visible
+        onResetCalibrationClicked: calibrationView.reset();
 
+        // debug options
         onShowPointCloud: arSample.showPointCloud(visible);
         onShowPlanes: arSample.showPlanes(visible);
 
-        onEmptySceneClicked: arSample.createEmptyScene();
-        onStreetsSceneClicked: arSample.createStreetsScene();
-        onImagerySceneClicked: arSample.createImageryScene();
-        onFullScaleTestSceneClicked: arSample.createFullScaleTestScene();
+        // full scale scenes
+        onEmptySceneClicked: {
+            setCalibrationFactors(0.000001);
+            arSample.createEmptyScene();
+        }
+        onStreetsSceneClicked: {
+            setCalibrationFactors(0.000001);
+            arSample.createStreetsScene();
+        }
+        onImagerySceneClicked: {
+            setCalibrationFactors(0.000001);
+            arSample.createImageryScene();
+        }
+        onFullScaleTestSceneClicked: {
+            setCalibrationFactors(0.000001);
+            arSample.createFullScaleTestScene();
+        }
 
-        onPointCloudSceneClicked: arSample.createPointCloudScene();
-        onYosemiteSceneClicked: arSample.createYosemiteScene();
-        onBorderSceneClicked: arSample.createBorderScene();
-        onBrestSceneClicked: arSample.createBrestScene();
-        onBerlinSceneClicked: arSample.createBerlinScene();
-        onTabletopTestSceneClicked: arSample.createTabletopTestScene();
+        // tabletop scenes
+        onPointCloudSceneClicked: {
+            setCalibrationFactors(0.0001);
+            arSample.createPointCloudScene();
+        }
+        onYosemiteSceneClicked: {
+            setCalibrationFactors(0.0001);
+            arSample.createYosemiteScene();
+        }
+        onBorderSceneClicked: {
+            setCalibrationFactors(0.0001);
+            arSample.createBorderScene();
+        }
+        onBrestSceneClicked: {
+            setCalibrationFactors(0.00001);
+            arSample.createBrestScene();
+        }
+        onBerlinSceneClicked: {
+            setCalibrationFactors(0.0001);
+            arSample.createBerlinScene();
+        }
+        onTabletopTestSceneClicked: {
+            setCalibrationFactors(0.0000001);
+            arSample.createTabletopTestScene();
+        }
+
+        // helper for calibration factor
+        function setCalibrationFactors(factor) {
```

Comment:
This is not descriptive. There are four possible factors, and this helper changes two of them to this `factor` and resets the other two to something else.  How about something like this?

```qml
readonly property double altitudeAndHeadingDefault: 0.01
...
function updateCalibrationFactors(latLongfactor) {
    calibrationView.latitudeFactor = latLongfactor;
    calibrationView.longitudeFactor = latLongfactor;
    calibrationView.altitudeFactor = altitudeAndHeadingDefault;
    calibrationView.headingFactor = altitudeAndHeadingDefault;
}
```

Reply:
In fact, this const value can be removed.